accepted/2021/architecture-targeting.md

```diff
@@ -0,0 +1,466 @@
+
+Some people have asked that `dotnet publish` produces `-c release` apps by default. Great idea but out of scope for this document.
+
+### Enable RID targeting for all (appropriate) verbs
+
+Today, the CLI assumes that the implicit RID is the correct one to use for apphost creation, but is [wrong in some scenarios](https://github.com/dotnet/sdk/issues/17241) and does not provide an override.
+
+The primary scenario is enabling the native-architecture SDK to target an emulated architecture. For example, `dotnet run`, `dotnet test`, and `dotnet tool install` should all accept a RID to influence apphost creation, such as with the following commands:
+
+- `dotnet run -r osx-x64 --`.
+- `dotnet test -r win-x64`
+- `dotnet tool install -r win-x64`
+
+Note: `dotnet watch` is effectively a wrapper over `dotnet run` and `dotnet test` so would follow the same model as they do.
+
+Where appropriate, the SDK should encode a different implicit RID, such as `osx-x64` when targeting .NET Core 3.1 on Apple Silicon macOS machines with the Arm64 .NET 6 SDK. The SDK should understand that .NET Core 3.1 only supports x64 and not Arm64 for macOS. The x64 SDK should provide the same behavior in reverse.
+
+### Enable shorthand RID targeting syntax
+
+RIDs are a pain to use on a regular basis. Particularly for emulation, it would be very useful to have a shorthand model for just specifying the architecture. The operating system remains constant.
+
+Imagine building a .NET 6 app on a Mac Arm64 machine, day in, day out. Every hour or two or three, you like to run a quick test run on x64. It would be really nice to type the following (or some analogue to it):
+
+```bash
+dotnet test -a x64
+```
+
+Instead of:
+
+```bash
+dotnet test -r osx-x64
+```
+
+It would also be useful (and symmetrical) to have a shorthand for operating system. The primarily advantage with this addition is that users could specify architecture and operating system separately, and never need to learn the .NET RID concept. This is very similar to the [golang model](https://golang.org/cmd/go/#hdr-Compile_and_run_Go_program).
+
+None, either or both shorthand values could be specified and both, one or neither would use implicit values, respectively, as demonstrated in the following examples.
+
+- `dotnet build`
+- `dotnet build -a x64`
+- `dotnet build --os Linux`
+- `dotnet build -a arm64 --os osx`
+
+Assuming a Windows Arm64 SDK, the following RIDs would be used in these examples:
+
+- `win-arm64`
+- `win-x64`
+- `linux-x64`
```

Comment:
```suggestion
- `linux-arm64`
```
I would expect this to default to linux-arm64.

Reply:
You are right. Oops.